MiniApp/Classes/core/MiniAppInfoFetcher.swift

```diff
@@ -44,21 +50,31 @@ internal class MiniAppInfoFetcher {
 
     func getMiniAppMetaInfo(miniAppId: String, miniAppVersion: String, apiClient: MiniAppClient, completionHandler: @escaping (Result<MiniAppManifest, MASDKError>) -> Void) {
 
-        apiClient.getMiniAppMetaData(appId: miniAppId, versionId: miniAppVersion) { (result) in
-            switch result {
-            case .success(let responseData):
-                guard let decodeResponse = ResponseDecoder.decode(decodeType: MetaDataResponse.self,
-                    data: responseData.data) else {
-                    return completionHandler(.failure(.invalidResponseData))
+        guard let manifest = getCachedManifest(miniAppId: miniAppId, miniAppVersion: miniAppVersion) else {
```

Comment:
I don't think this is the right place handle the caching. This caching is being added because we want to avoid making multiple API calls for the same thing twice in row, so I think the caching should be added closer to the API functionality - i.e. in [`ManifestDownloader`](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/core/ManifestDownloader.swift).

So basically `ManifestDownloader` will cache just a single manifest (the last one which was fetched) and also which mini app id/version it is for. So the next time that a request is made to `ManifestDownloader`, it will check whether it's the same Manifest which was just fetched, and, if so, then immediately return the cached manifest.

Reply:
> we want to avoid making multiple API calls for the same thing twice in row

does this mean we would only need to store it in a variable of the MiniApp singleton?

Reply:
@corycaywood [ManifestDownloader](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/core/ManifestDownloader.swift) is meant for list of manifest files to download for the mini app. But this ticket is meant to cache the meta-data right ?

Reply:
@corycaywood [ManifestDownloader](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/core/ManifestDownloader.swift) is meant for list of manifest files to be downloaded for MiniApp. But this ticket is for meta-data caching right ?

Reply:
@leorjoseph Oh sorry, yeah that's right. Then maybe we need to create a `MetaDataDownloader` class to encapsulate this functionality?

@Climbatize Yeah, I guess we could just cache the response in memory. But the manifest.json could be very long for some mini apps, so I'm not sure whether we should do that. What do you think?

Reply:
Done